mbed_targets/mbed_targets.py

```diff
@@ -53,22 +52,9 @@ class MbedTargets:
 
     def __init__(self) -> None:
         """Retrieve target data from the online database."""
-        self._target_data: Optional[
-            List[dict]] = target_database.get_target_data()
+        self._target_data: List[dict] = target_database.get_target_data()
```

Comment:
This change exposes the current caller to the API of this package to the new exceptions without documentation.

Reply:
I'll add a comment to the class docstring to clarify which exceptions it can throw.